src/main/java/com/google/sps/storage/DatastoreGuides.java

```diff
@@ -64,7 +63,12 @@ public Guide put(Guide guide){
         return listFromQueryResults(datastore.run(query));
     }
 
-    public Guide get(long id){
-        return toGuide(datastore.get(keyFactory.newKey(id)));
+    public Optional<Guide> get(long id){
+        Entity entity = datastore.get(keyFactory.newKey(id));
+        if(entity == null){
```

Comment:
This could also be `Optional.ofNullable(entity).map(this::toGuide)`;

Reply:
This is so clean! Changed in next commit.